### ISSUE TITLE
Problem running RefFreeDMA test

### ISSUE THREAD

**Issue by srialle** [jklughammer/RefFreeDMA#2]
Dear Johanna,

I am trying to use RefFreeDMA for a project on a non-model species, and your method seems very promising.

Unfortunately I can't make the test work. I followed the steps you mention in the Quick Start and I get the following message : 

> $ /data/software/RefFreeDMA-master/RefFreeDMA.sh $testDir/RefFreeDMA_test/meta/RefFreeDMA_test.cfg
> 
> Checking for required python libraries...
> python Bio ... OK
> python pysam ... OK
> python bitarray ... OK
> python guppy ... OK
> 
> 
> Checking for required tools...
> samtools ... OK
> bedtools ... OK
> trim_galore ... OK
> cutadapt ... OK
> bowtie2 ... OK
> bsmap ... OK
> samToFastq.jar ... OK
> 
> -------Read preparation-------
> 12 samples selected for reference generation:
> |Hum_1_G||Hum_1_L||Hum_1_M||Hum_2_G||Hum_2_L||Hum_2_M||Hum_3_G||Hum_3_L||Hum_3_M||Hum_4_G||Hum_4_L||Hum_4_M|
> Unconverted tag: no unconv_tag
> BSF_0116_C55NLACXX_3__Hum_2_L
> submitted
> 
> real	0m0.001s
> user	0m0.001s
> sys	0m0.000s
> BSF_0116_C55NLACXX_3__Hum_2_M
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0116_C55NLACXX_3__Hum_3_M
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0117_C5578ACXX_5__Hum_1_L
> submitted
> 
> real	0m0.001s
> user	0m0.001s
> sys	0m0.000s
> BSF_0117_C5578ACXX_5__Hum_1_M
> submitted
> 
> real	0m0.001s
> user	0m0.001s
> sys	0m0.000s
> BSF_0117_C5578ACXX_5__Hum_4_G
> submitted
> 
> real	0m0.001s
> user	0m0.001s
> sys	0m0.001s
> BSF_0117_C5578ACXX_5__Hum_4_M
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0117_C5578ACXX_6__Hum_2_G
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0117_C5578ACXX_6__Hum_4_L
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0121_C5CTMACXX_7__Hum_3_G
> submitted
> 
> real	0m0.001s
> user	0m0.001s
> sys	0m0.000s
> BSF_0121_C5CTMACXX_8__Hum_1_G
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> BSF_0121_C5CTMACXX_8__Hum_3_L
> submitted
> 
> real	0m0.001s
> user	0m0.000s
> sys	0m0.001s
> Didn't find the expected number of uniq.ref files (12) or trimmed.fq files (12). Exiting!`

I am working on CentOS 7 and I use the tools you provide. 

Do you have an idea of what is going wrong?

Please tell me if you need any further information.

Many thanks,

Stephanie

**Comment by jklughammer** [comment]
Hi Stephanie, 
I will look into this. Did RefFreeDMA produce any log files with informative error message? 
The log files should be in $testDir/RefFreeDMA_test/log.
Could you please post any messages that seem relevant?
Thanks , Johanna

**Comment by jklughammer** [comment]
Thinking about it, I have a strong suspicion what might be the problem: Could it be that you do not have java installed? RefFreeDMA doesn't explicitly test this so it wouldn't show in the pre-flight tests. 
I also just tested on Ubuntu and everything worked, so if my java guess is wrong, there might be an incompatibility with CentOS.
Best, Johanna

**Comment by srialle** [comment]
Hi Johanna,

Thank you for your response.

I had to change the code in your function get_proc_stats (line 113 in RefFreeDMA.sh) because the command /usr/bin/time was not recognized on my system. I replaced /usr/bin/time by time : 
`eval "time -o $logdir/time --append -f 'command: %C\ntime(sec): %e(real_time) %S(system) %U(user)\nmemory(Kbytes): %M(peak) %K(avg_total)\n' $1"`
And now I get the following error in files like prepareReads_BSF_0116_C55NLACXX_3__Hum_2_L.log :
 `/data/software/RefFreeDMA-master/RefFreeDMA.sh: ligne113: -o : commande introuvable`, which means "line 113: -o command not found".
The function time in CentOS might take different parameters than in Ubuntu, don't you think?

Regarding Java, it is installed on my system. We recently had a problem (with another software) with some Java libraries we were lacking and had to install, but we got an error message leading us to the solution.

Thanks again,

Best,

Stephanie

**Comment by srialle** [comment]
Hi again,
It was indeed because of the time command that it was not working. I have suppressed this part and now it is running, with no problem so far. The function is now:

> function get_proc_stats {
> echo $2 >> $logdir/time
> eval "$1"
> 
>}

I won't have the time measurements but at least it will work :).

Thanks again for your time.

Best,

Stephanie

**Comment by jklughammer** [comment]
Hi Stephanie,
glad it is working now. I chose to use the GNU time command which (on ubuntu) can be accessed by specifying the path to the binary. This time command indeed has different options and functionalities to the other time commands. Maybe on your system the GNU time command is not available or in a different location. 
In any case, as you correctly note you don't need the timing and memory profiling I intended with this function - it is just "nice to have".
Best, Johanna
